src/dali/mapped_graph.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2023 Neal Chambers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import datetime, timedelta
+from typing import Dict, Iterator, List, NamedTuple, Optional, Set
+
+from prezzemolo.graph import Graph
+from prezzemolo.utility import ValueType
+from prezzemolo.vertex import Vertex
+from rp2.rp2_decimal import RP2Decimal
+from rp2.rp2_error import RP2TypeError
+
+from dali.historical_bar import HistoricalBar
+from dali.logger import LOGGER
+
+
+class Alias(NamedTuple):
+    from_asset: str
+    to_asset: str
+
+
+# Hard-coded base aliases
+_UNIVERSAL_ALIASES = {
+    Alias(from_asset="LUNA", to_asset="LUNC"): RP2Decimal("1"),
+    Alias(from_asset="XBT", to_asset="BTC"): RP2Decimal("1"),
+}
+
+_EXCHANGE_SPECIFIC_ALIASES = {
+    "Coinbase Pro": {
+        Alias(from_asset="ETH2", to_asset="ETH"): RP2Decimal("1"),
+    },
+    "Pionex": {
+        Alias(from_asset="MBTC", to_asset="BTC"): RP2Decimal("0.001"),
+        Alias(from_asset="METH", to_asset="ETH"): RP2Decimal("0.001"),
+    },
+}
+
+_UNIVERSAL: str = "UNIVERSAL"
+
```

Comment:
Well, `ccxt.py` is getting too big. I'd like to avoid putting anything more than what is absolutely necessary in it. `abstract_pair_converter.py` is mostly `MappedGraph` agnostic at this point and it should probably stay that way in the event someone else wants to use it to build a pair converter plugin that doesn't make use of a graph. That leaves `MappedGraph` or a subclass as you say. See my more general comment.